packages/transports/rest/src/directives/httpOperation.ts

```diff
@@ -321,61 +326,66 @@ export function addHTTPRootFieldResolver(
       } else if (response.status === 204 || (response.status === 200 && responseText === '')) {
         responseJson = {};
       } else if (response.status.toString().startsWith('2')) {
-        logger.debug(`Unexpected response in ${field.name};\n\t${responseText}`);
-        return createGraphQLError(`Unexpected response in ${field.name}`, {
-          extensions: {
-            http: {
-              status: response.status,
-              statusText: response.statusText,
-              headers: getHeadersObj(response.headers),
-            },
-            request: {
-              url: fullPath,
-              method: httpMethod,
-            },
-            responseText,
-            originalError: {
-              message: error.message,
-              stack: error.stack,
-            },
+        logger.debug(`Unexpected upstream HTTP response in ${field.name};\n\t${responseText}`);
+        const extensions: MeshUpstreamErrorExtensions = {
+          subgraph: sourceName,
+          http: {
+            status: response.status,
+            statusText: response.statusText,
+            headers: getHeadersObj(response.headers),
+          },
+          request: {
+            endpoint: fullPath,
+            method: httpMethod,
           },
+          responseBody: responseText,
+        };
```

Comment:
Since this seems duplicated in the following changes, perhaps we should have a function ?

```suggestion
        const extensions = makeHttpErrorExtension(response, fullPath, httpMethod, responseText)
```